client/layout/logged-out.jsx

```diff
@@ -50,6 +56,18 @@ const LayoutLoggedOut = ( {
 	isPartnerSignupStart,
 	locale,
 } ) => {
+	const dispatch = useDispatch();
+	const isStateRandomlyCleared = useSelector( getIsStateRandomlyCleared );
+
+	useEffect( () => {
```

Comment:
Same as above in terms of abstracting away from the component. Even if it is not async loaded (since logged-out is used in SSR), from the perspective of code organization, readability and maintainability standpoint.